src/c++/perf_analyzer/client_backend/triton/triton_client_backend.cc

```diff
@@ -362,6 +369,83 @@ TritonClientBackend::ModelInferenceStatistics(
   return Error::Success;
 }
 
+Error
+TritonClientBackend::Metrics(triton::perfanalyzer::Metrics& metrics)
+{
+  try {
+    std::string metrics_endpoint_text{""};
+    AccessMetricsEndpoint(metrics_endpoint_text);
+    ParseAndStoreMetrics(metrics_endpoint_text, metrics);
+  }
+  catch (const std::exception& e) {
```

Comment:
I think it is best practice to not catch all the exceptions. Only catch PerfAnalyzerException and return the error code included in it.

Reply:
Switched to `PerfAnalyzerException` 👍